tests/cloner_test.go

```diff
@@ -1813,45 +1823,57 @@ var _ = Describe("all clone tests", func() {
-			Expect(err).ToNot(HaveOccurred())
-			f.ForceBindPvcIfDvIsWaitForFirstConsumer(targetDv1)
-
-			By("Cloning from the source DataVolume to target2 in parallel")
-			targetDv2 = utils.NewDataVolumeForImageCloning("target-dv2", "200Mi", f.Namespace.Name, sourceDv.Name, sourceDv.Spec.PVC.StorageClassName, sourceDv.Spec.PVC.VolumeMode)
-			targetDv2, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv2)
-			Expect(err).ToNot(HaveOccurred())
-			f.ForceBindPvcIfDvIsWaitForFirstConsumer(targetDv2)
-
-			By("Cloning from the source DataVolume to target3 in parallel")
-			targetDv3 = utils.NewDataVolumeForImageCloning("target-dv3", "200Mi", f.Namespace.Name, sourceDv.Name, sourceDv.Spec.PVC.StorageClassName, sourceDv.Spec.PVC.VolumeMode)
-			targetDv3, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv3)
-			Expect(err).ToNot(HaveOccurred())
-			f.ForceBindPvcIfDvIsWaitForFirstConsumer(targetDv3)
+			// By not waiting for completion, we will start 3 transfers in parallel
+			By("Cloning #NumOfClones times in parallel")
+			for i := 1; i <= NumOfClones; i++ {
+				By("Cloning from the source DataVolume to target" + strconv.Itoa(i))
+				targetDv := utils.NewDataVolumeForImageCloning("target-dv"+strconv.Itoa(i), "200Mi", f.Namespace.Name, sourceDv.Name, sourceDv.Spec.PVC.StorageClassName, sourceDv.Spec.PVC.VolumeMode)
+				targetDv.Annotations[controller.AnnPodRetainAfterCompletion] = "true"
+				targetDv, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv)
+				Expect(err).ToNot(HaveOccurred())
+				f.ForceBindPvcIfDvIsWaitForFirstConsumer(targetDv)
+				targetDvs = append(targetDvs, targetDv)
+			}
 
-			dvs := []*cdiv1.DataVolume{targetDv1, targetDv2, targetDv3}
-			for _, dv := range dvs {
+			podsNodeName := make(map[string]string)
+			for _, dv := range targetDvs {
 				By("Waiting for clone to be completed")
 				err = utils.WaitForDataVolumePhaseWithTimeout(f, f.Namespace.Name, cdiv1.Succeeded, dv.Name, 3*90*time.Second)
 				Expect(err).ToNot(HaveOccurred())
 			}
 
-			for _, dv := range dvs {
+			for _, dv := range targetDvs {
 				By("Verifying MD5 sum matches")
 				matchFile := filepath.Join(testBaseDir, "disk.img")
 				Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(dv), matchFile, md5sum[:32])).To(BeTrue())
+
+				pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dv.Namespace).Get(context.TODO(), dv.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				if cloneSourcePod := pvc.Annotations[ctrl.AnnCloneSourcePod]; cloneSourcePod != "" {
+					By(fmt.Sprintf("Getting pod %s/%s", dv.Namespace, cloneSourcePod))
+					pod, err := f.K8sClient.CoreV1().Pods(dv.Namespace).Get(context.TODO(), cloneSourcePod, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					podsNodeName[dv.Name] = pod.Spec.NodeName
```

Comment:
Would it make sense to have the map be `map[string]bool` and on this line just say `podsNodeName[pod.Spec.NodeName]=true` and below check `Expect(podsNodeName).To(HaveLen(1))` We are essentially creating a set of node names, and ensure there is only 1 node name. This demonstrates all the pods were on the same node.